neqo-transport/src/recovery.rs

```diff
@@ -904,9 +907,13 @@ impl LossRecovery {
 
     fn fire_pto(&mut self, pn_space: PNSpace, allow_probes: PNSpaceSet) {
         if let Some(st) = &mut self.pto_state {
-            st.pto(pn_space, allow_probes);
+            st.pto(pn_space, allow_probes, &mut *self.stats.borrow_mut());
         } else {
-            self.pto_state = Some(PtoState::new(pn_space, allow_probes));
+            self.pto_state = Some(PtoState::new(
+                pn_space,
+                allow_probes,
+                &mut *self.stats.borrow_mut(),
```

Comment:
Save on duplication and add a function to PtoState that accumulates stats like:

```rust
self.pto_state.as_mut().unwrap().count_pto(&mut *self.stats.borrow_mut());
```
Then you don't count in two places and invoke in two places.  The only icky bit is the unwrap() call.